frontend_tests/node_tests/message_list_view.js

```diff
@@ -689,6 +690,71 @@ test("merge_message_groups", () => {
     })();
 });
 
+test("build_message_groups", () => {
+    function build_message(content, id) {
+        stream_data.add_sub({stream_id: 13, name: "test"});
+
+        return {
+            msg: {
+                clean_reactions: new Map(),
+                collapsed: false,
+                content,
+                display_recipient: "Zoolippy",
+                id,
+                is_stream: true,
+                recipient_id: 35,
+                sender_full_name: "King Hamlet",
+                sender_id: 10,
+                sender_email: "hamlet@zulip.com",
+                stream: "test",
+                stream_id: 13,
+                timestamp: 1646398167,
+                topic: "fancy topic",
+                type: "stream",
+                unread: false,
+                url: `http://zulip.zulipdev.com/#narrow/stream/13-test/topic/fancy.20topic/near/${id}`,
+            },
+            stream_url: "#narrow/stream/13-test",
+            timestr: "4:49 AM",
+            topic_url: "#narrow/stream/13-test/topic/fancy.20topic",
+        };
+    }
+
+    (function test_collapse_messages_true() {
```

Comment:
I would consider flattening out the lexical structure here a bit, and break these into separate tests with the test wrapper:

~~~
function build_message(...) {}

test("collapse", ...)
test("don't collapse", ..)

Reply:
I think if I did that, the tests would be structured differently than the other tests in the file - is that okay?

Reply:
Yes

Reply:
Just to elaborate a bit on this, the old IIFE code structure dates back to pre-ES6 days, and it may even date back to when our `run_test` wrapper wasn't so nice.  And in some cases I think the original authors didn't have an appreciation for "flatter is better" or small functions.  I also think I was originally responsible for introducing this pattern, and I regret it.

Long story short, unless you see benefits here that I no longer see, we should move away as quickly from the IIFE idiom as possible.

Reply:
My preference would be to move to the new format when we create new test files, or to update entire files at a time -- I think it's confusing to contributors to have multiple styles in the same file to pick between.

If you'd like to pick a day in the near future where we spend some time together converting a few test files, I'd be down to work on that with you :)

Reply:
Sure, makes sense.